### PR TITLE
Export flags and textvars in codebooks

### PR DESCRIPTION
**Pull request by jornvandebeek**

### CODE REVIEW

File: lib/quby/questionnaires/entities/textvar.rb [hunk 1]
```diff
@@ -8,6 +8,13 @@ def initialize(key:, description:, default: nil)
           super(key, description, default)
         end
         # rubocop:enable ParameterLists
+
+        def to_codebook(_options = {})
+          output = []
+          output << "#{key} Text variabele"
```
Comment (timraasveld) [inline lib/quby/questionnaires/entities/textvar.rb]: Tekstvariabele

File: lib/quby/questionnaires/entities/flag.rb [hunk 2]
```diff
@@ -18,6 +18,16 @@ def initialize(key:,
         def if_triggered_by(answer_flags)
           yield if answer_flags[key] == trigger_on
         end
+
+        def to_codebook(_options = {})
+          output = []
+          output << "#{key} flag"
+          output << " 'true' - #{description_true}"
+          output << " 'false' - #{description_false}"
+          output << " '' (leeg) - Vlag niet ingesteld, informatie onbekend"
```
Comment (jornvandebeek) [inline lib/quby/questionnaires/entities/flag.rb]: Het lijkt me handiger als de codeboeken niet proberen te omschrijven wat de vlag voor uitwerking heeft, maar zich enkel bemoeien met wat voor informatie de vlag voor stelt.

Comment (hampei) [inline lib/quby/questionnaires/entities/flag.rb]: Zet dan een `(default)` achter 1 van de waardes. Want het niet zetten van een vlag is niet echt een edge-case ofzo, en dan wil je dus wel weten hoe die wordt geïnterpreteerd.

**Comment by hampei** [comment]
Falende test

File: lib/quby/questionnaires/entities/questionnaire.rb [hunk 3]
```diff
@@ -183,9 +181,20 @@ def to_codebook(options = {})
             output << ""
           end
 
+          flags.each do |_flag_key, flag|
```
Comment (jornvandebeek) [inline lib/quby/questionnaires/entities/questionnaire.rb]: Ik laat het toch zo.

File: lib/quby/questionnaires/entities/flag.rb (hunk 2 above)
Comment (jornvandebeek) [inline lib/quby/questionnaires/entities/flag.rb]: Volgens mij is dat niet nodig, de default is altijd het lege geval. De trigger_on heeft geen invloed op de beginwaarde.

File: lib/quby/questionnaires/entities/questionnaire.rb (hunk 3 above)
Comment (timraasveld) [inline lib/quby/questionnaires/entities/questionnaire.rb]: Idd lelijk, maar het is nu (helaas) de conventie. Lijkt me fijner om het in een aparte refactor-PR te doen zodat we één standaard houden

File: lib/quby/questionnaires/entities/flag.rb (hunk 2 above)
Comment (hampei) [inline lib/quby/questionnaires/entities/flag.rb]: Je wilt zeggen dat `flag.trigger_on == true && flag.value == false` niet exact dezelfde lijst geeft als `flag.value == nil` ?

Of dat dit voor de gebruiker niet belangrijke informatie is om te weten dat je die instanties hetzelfde kan behandelen?

File: lib/quby/questionnaires/entities/questionnaire.rb (hunk 3 above)
Comment (hampei) [inline lib/quby/questionnaires/entities/questionnaire.rb]: Waar staat onze conventie over hele lange functies maken en nooit subfuncties maken gedocumenteerd?

Comment (jornvandebeek) [inline lib/quby/questionnaires/entities/questionnaire.rb]: Nou het is ook een snelheidsding, het voelt gewoon vies om een boel `*blah.map{}.flatten` in een grote array te zetten. De << (streaming/appending) aanpak is de correcte, ik zal het nog herschrijven.

Comment (hampei) [inline lib/quby/questionnaires/entities/questionnaire.rb]: Het is geen snelheidsding, flatten is zelfs net iets sneller, maar volledig verwaarloosbaar. 
Het is idd jammer van de `.map { |s| s + "\n" }`, zonder dat was het perfect geweest, in 1x een high-level overzicht van wat er in komt. 
Wat mij betreft 10x leesbaarder/scanbaarder dan de appending methode en veel meer railsy. 
Maar wil best toegeven dat dat een mening is en dat andere aanpakken ook best leesbaar kunnen zijn.

File: lib/quby/questionnaires/entities/flag.rb (hunk 2 above)
Comment (jornvandebeek) [inline lib/quby/questionnaires/entities/flag.rb]: Ik probeer dat niet te zeggen, je stelling is waar.  Het is een momentopname. Stel dat je een flag maakt 'client rookt', trigger on true shows hidden by default rookvragen, maar de lijst verandert op een gegeven moment naar trigger on false, hides rookvragen (de correcte implementatie omdat je de rookvragen ook in het nil geval wil laten zien). Dan verandert dat 'default' gegeven in de codebook. Verder suggereert default dat het kiezen om het leeg te maken een bewuste keuze is, in plaats van dat dat eigenlijk de default is. Het trigger_on gedrag is een iets te implementatiespecifiek gegeven, ik laat het liever aan de codeboek/export gebruiker over om uit de invullingen zelf icm de vlaginhoud af te leiden of alle verwachte vragen zijn beantwoord.
